src/ToDoList.js

```diff
@@ -18,107 +16,97 @@ import {
-  textDecoration: "line-through",
-  textDecorationStyle: "solid",
-  textDecorationThickness: "2px",
-  color: "black",
-};
-
-/*const deleteButtonStyle = {
-  fontSize: "10px",
-  padding: "3px 3px",
-};*/
+const containerStyle = css`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  min-height: 100vh;
+  flex-direction: column;
+`;
+
+const titleStyle = css`
+  color: black;
+  font-size: 24px;
+  width: 300px;
+  height: 50px;
+  border-radius: 20px;
+  background-color: lightBlue;
+  margin: 0 auto;
+  font-weight: bold;
+  margin-top: 30px;
+  display: flex;
+  flex-direction: column;
+`;
+
+const inputContainerStyle = css`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  margin-top: 30px;
+`;
+
+const completedTaskStyle = css`
+  text-decoration: line-through;
+  text-decoration-style: solid;
+  text-decoration-thickness: 2px;
+  color: black;
+`;
 
 const ToDoList = () => {
+  const [tasks, setTasks] = useState([]);
+  const [taskInfo, setTaskInfo] = useState("");
+  const [filter, setFilter] = useState("all");
+
   useEffect(() => {
     const storedTasks = localStorage.getItem("tasks");
     if (storedTasks) {
       setTasks(JSON.parse(storedTasks));
     }
   }, []);
 
-  const [tasks, setTasks] = useState(() => {
-    const storedTasks = localStorage.getItem("tasks");
-    return storedTasks ? JSON.parse(storedTasks) : [];
-  });
-
-  const [taskInfo, setTaskInfo] = useState("");
-  const [filter, setFilter] = useState("all");
-
-  useEffect(() => {
-    localStorage.setItem("tasks", JSON.stringify(tasks));
-  }, [tasks]);
-
   const handleInputChange = (event) => {
     setTaskInfo(event.target.value);
   };
+
   const addTask = () => {
     if (taskInfo.trim() !== "") {
+      const isTaskExists = tasks.some((task) => task.task === taskInfo.trim());
+
+      if (isTaskExists) {
+        alert("Task already exists!");
+        return;
+      }
+
       const newTask = {
         id: Date.now(),
-        task: taskInfo,
+        task: taskInfo.trim(),
         completed: false,
       };
 
-      setTasks([...tasks, newTask]);
+      const updatedTasks = [...tasks, newTask];
+      setTasks(updatedTasks);
       setTaskInfo("");
+      localStorage.setItem("tasks", JSON.stringify(updatedTasks));
     }
   };
 
   const deleteTask = (taskId) => {
     const updatedTasks = tasks.filter((task) => task.id !== taskId);
     setTasks(updatedTasks);
+    localStorage.setItem("tasks", JSON.stringify(updatedTasks));
   };
 
   const markAsCompleted = (taskId) => {
     const updatedTasks = tasks.map((task) => {
       if (task.id === taskId) {
-        return { ...task, completed: !task.completed };
+        const updatedTask = { ...task, completed: !task.completed };
+        return updatedTask;
       }
       return task;
     });
+
     setTasks(updatedTasks);
+    localStorage.setItem("tasks", JSON.stringify(updatedTasks));
   };
 
   const filterTasks = () => {
```

Comment:
We don't use `switch case`, but `if return`.

```
if (filter === "completed") return tasks.filter((task) => task.completed);
if (filter === "toDo") return tasks.filter((task) => !task.completed);
return tasks;
```

Reply:
Done.